docs/config/version_provider.md

```diff
@@ -0,0 +1,308 @@
+# Version Providers
+
+Version providers are the mechanism by which Commitizen reads and writes version information in your project. They abstract away the details of where and how version numbers are stored, allowing Commitizen to work seamlessly with different project types and package management systems.
+
+## Overview
+
+By default, Commitizen uses the `commitizen` provider, which stores the version in your Commitizen configuration file. However, you can configure Commitizen to use any available provider that matches your project's setup. This is particularly useful when you want Commitizen to manage versions in the same location as your package manager (e.g., `package.json` for Node.js projects, `pyproject.toml` for Python projects).
+
+## Built-in Providers
+
+Commitizen includes several built-in version providers for common package management formats:
+
+### `commitizen` (Default)
+
+The default version provider stores and retrieves the version from your Commitizen configuration file (e.g., `pyproject.toml`, `.cz.toml`, etc.).
+
+**Use when:**
+- You want to keep version management separate from your package manager
+- Your project doesn't use a standard package manager
+- You need maximum flexibility in version management
+
+**Configuration:**
+```toml
+[tool.commitizen]
+version_provider = "commitizen"
+version = "0.1.0"  # Required when using this provider
+```
+
+### `scm`
+
+Fetches the version from Git tags using `git describe`. This provider **only reads** version information and never writes it back to files. It's designed to work with tools like `setuptools-scm` or other package manager `*-scm` plugins that derive version numbers from Git history.
+
+**Use when:**
+- You're using `setuptools-scm` or similar tools
+- You want version numbers derived from Git tags
+- You don't want Commitizen to modify any files for version management
+
+**Configuration:**
+```toml
+[tool.commitizen]
+version_provider = "scm"
+# No version field needed - it's read from Git tags
+```
+
+!!! note
+    The `scm` provider is read-only. When you run `cz bump`, it will create a Git tag but won't update any files. This is intentional and works well with tools that derive versions from Git tags.
+
+### `pep621`
+
+Manages version in `pyproject.toml` under the `project.version` field, following [PEP 621](https://peps.python.org/pep-0621/) standards.
+
+**Use when:**
+- You're using a modern Python project with PEP 621-compliant `pyproject.toml`
+- You want version management integrated with your Python project metadata
+
+**Configuration:**
+```toml
+[tool.commitizen]
+version_provider = "pep621"
+```
+
+**Example `pyproject.toml`:**
+```toml
+[project]
+name = "my-package"
+version = "0.1.0"  # Managed by Commitizen
+```
+
+### `poetry`
```

Comment:
I think this only make sense for poetry < 2.0. for poetry >= 2.0, pep621 is suggested